openquake/hazardlib/gsim/chao_2020.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2015-2018 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`ChaoEtAl2020SInter`
+               :class:`ChaoEtAl2020SSlab`
+               :class:`ChaoEtAl2020Asc`
+"""
+import math
+
+import numpy as np
+
+from openquake.hazardlib import const
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib.imt import PGA, PGD, PGV, SA
+
+
+class ChaoEtAl2020SInter(GMPE):
+    """
+    Chao et al. (2020) for Subduction Interface.
+    """
+
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTERFACE
+
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([PGA, PGD, PGV, SA])
+
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT
+    ])
+
+    REQUIRES_DISTANCES = {'rrup'}
+
+    REQUIRES_RUPTURE_PARAMETERS = {'mag', 'ztor'}
+
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'vs30measured', 'z1pt0'}
```

Comment:
Would be good adding also the reference rock conditions as done here: https://github.com/gem/oq-engine/blob/master/openquake/hazardlib/gsim/abrahamson_2014.py#L82

Reply:
fixed